### PR TITLE
Adds quick start new visuals for quick start card

### LINKED ISSUE #16587: Quick Start: New visuals for Quick Start card
Parent #16347

## Description

Visual changes to the Quick Start card.

Design ref: pbArwn-4wS-p2

## Tasks

- [x] Update Quick Start card 
    - [x] Update to new designs for Existing Site only
    - [x] Progress bar (states: in progress, complete)

### PR DESCRIPTION
**Pull request by AjeshRPai**
Parent #16587 

This PR Updates the Quick Start card to new designs for the Existing Site tours (New Site tours are using the original designs)

## WordPress app

|Quick start Finished |  Quick start Finished(Dark mode)| 
|---|---|
|||

---
#### Existing Site tours

1. Login to an app with existing site tour quick start in progress 
3. Verify: The quick start card design has been updated
4. Complete a tour
5. Verify: The progress bar updates accordingly
6. Complete all tours
7. Verify: The progress bar turns green and a checkmark icon appears 

#### New Site tours
1. Login to an app with old quick start tour in progress
2. Verify: The quick start card design hasn't changed

### Merge Instructions 
- [x] Revert the commit [128c170](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/128c170883499d1dcffd6f8b1733f70e5e1fccb9). 
- [x] Remove not ready for merge label 
- Merge

## Regression Notes
1. Potential unintended areas of impact
 Quick start card not being shown for existing users and new users

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests 

4. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AjeshRPai** [comment]
@ashiagr @osullivanchris 
As the issue #https://github.com/wordpress-mobile/WordPress-Android/issues/16615 exists, the quick start card will disappear as soon as the quick start is completed. In order to do the design review of the completed state of the quick start card, I have disabled the removal of the quick start card in the completed state with the commit [128c170](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/128c170883499d1dcffd6f8b1733f70e5e1fccb9). This will be reverted once the design review is done and @osullivanchris has approved the changes.

File: WordPress/src/main/res/layout/new_quick_start_task_type_item.xml
Comment (ashiagr) [inline WordPress/src/main/res/layout/new_quick_start_task_type_item.xml]: Np: This part can be formatted.

File: WordPress/src/main/res/layout/new_quick_start_task_type_item.xml
Comment (ashiagr) [inline WordPress/src/main/res/layout/new_quick_start_task_type_item.xml]: Np: Hardcoded margins can be replaced with values from `dimens.xml`.

File: WordPress/src/main/res/layout/new_quick_start_task_type_item.xml
Comment (ashiagr) [inline WordPress/src/main/res/layout/new_quick_start_task_type_item.xml]: Np: This illustration image is actually not an icon and can be renamed. Wdyt?

**Review by ashiagr — COMMENTED** [review]
Works as described!  Left a few nitpicks for you to consider.

Also, 
- Looks like you deleted `checkstyle-idea.xml.` Can you bring it back?
- Not sure what we decided about the icon on the card toolbar, and if it should be present for both `home` and `menu` tabs. I'll leave that discussion to you and Chris.

**Comment by ashiagr** [comment]
👋  @AjeshRPai 

Two more things: 
1. We used the same illustration in this PR and in the qs tasks list modal new visuals available in `trunk` now. They are included twice with different names. Can we merge trunk to this PR and get rid of one of them?
2. Let's update the release note to include one item for quick start new visuals. We can include both PRs in it.

**Comment by AjeshRPai** [comment]
Hey @ashiagr , Thanks for pointing out the changes

>We used the same illustration in this PR and in the qs tasks list modal new visuals available in trunk now. They are included twice with different names. Can we merge trunk to this PR and get rid of one of them?

Done in https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/1d2dc8eaaf4db278b18b443d403bbe90a01ddf84
1. I removed both the files from your PR
2. renamed the image file to `ic_quick_start_tour_illustration` for better clarity
3. Update the usage in card and in quick start list 

>Let's update the release note to include one item for quick start new visuals. We can include both PRs in it.

Done in [962f145](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/962f1456ee8d8cfc5d53ab8ce3e7001097966c2d)

**Comment by ashiagr** [comment]
Thanks @AjeshRPai!

> renamed the image file to ic_quick_start_tour_illustration for better clarity

Can we rename it to `img_quick_start_tour_illustration` as it is not an icon? 😅

**Comment by AjeshRPai** [comment]
@osullivanchris, I have a few questions  

Title 
- Does it need to be in two lines?. 
- Title string should be "Get to know the app"  or "Get to know the App"

Subtitle/Secondary Text(1 out of 6 Complete) 
- Suggestion: This looks a little bit more separated from the title with 16 dp margin-top, should we make it 8 dp?. 

Illustration 
- I have given the following size right now 
	- 54dp width hardcoded
	- height - top to the title top and bottom to the progress bar bottom
Is this fine?

Task completed Icon
- Can u share with me the size of the task completed tick mark shown. Right now the size is 24dpX24dp

❓  Is the design or specs of the card different in the home and menu tab?

File: WordPress/src/main/res/layout/new_quick_start_task_type_item.xml
Comment (AjeshRPai) [reply WordPress/src/main/res/layout/new_quick_start_task_type_item.xml]: @ashiagr I have done the formatting in the commit. https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/9782625feca58e29b926820934e929a33d627cbd.

File: WordPress/src/main/res/layout/new_quick_start_task_type_item.xml
Comment (AjeshRPai) [reply WordPress/src/main/res/layout/new_quick_start_task_type_item.xml]: 👍🏼 Done in https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/9782625feca58e29b926820934e929a33d627cbd

File: WordPress/src/main/res/layout/new_quick_start_task_type_item.xml
Comment (AjeshRPai) [reply WordPress/src/main/res/layout/new_quick_start_task_type_item.xml]: 👍🏼 I have renamed it to `quick_start_item_img` in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/9782625feca58e29b926820934e929a33d627cbd, let me know if you have any other suggestions. 🤔

**Comment by AjeshRPai** [comment]
@ashiagr 

>Looks like you deleted checkstyle-idea.xml. Can you bring it back?

This was due to a setting in my IDE, sorry. I have reverted the deletion.

**Comment by ashiagr** [comment]
Changes look good to me @AjeshRPai 🙇‍♀️ 
I'll approve the PR once https://github.com/wordpress-mobile/WordPress-Android/pull/16629#issuecomment-1138262671 is addressed. Good job.

**Review by osullivanchris — CHANGES_REQUESTED** [review]
👋 Hey Ajesh. Thanks for the ping. The build is looking much better, and happy to answer the clarifying questions.

I have a couple of small details first:
- the heading text should be the same size as the 'Create your first/next post' heading text. Right now it looks much larger.
- The progress bar would look better 1dp larger in height
- Changing from the specs, I think it will look better if there's 32dp padding between the left content area (heading, subheading, progress bar) and the illustration on the right. At the moment the progress bar looks too close to the illustration. 

Now for your questions

> Does it need to be in two lines?.

No I don't think we would ever make a rule like that. It wouldn't be very good semantically. Although I do think the balance of the card is nice when it goes to two lines, we should let it fall whatever way it falls

> Title string should be "Get to know the app" or "Get to know the App"

Get to know the app

> Subtitle/Secondary Text(1 out of 6 Complete)...Suggestion: This looks a little bit more separated from the title with 16 dp margin-top, should we make it 8 dp?.

I see what you're saying but I think the space balances things vertically. Especially if 'Get to know the app' is only on one line on some devices, the content will be super short compared with the illustration. So I think we should keep 16. I also think its good for the progress bar and 'N out of N complete' to look more tight together than to the heading.

> Illustration size

54 width sounds right. Not sure I follow on the height. The original aspect ratio should be preserved so perhaps we should only define the width. For the vertical position, I was originally thinking centred to the content to the left. Maybe this is what you mean. Vertically centred between the title and the progress bar. 

Another thought I have on the height is now that we're seeing it looks quite different if the text wraps or not, whether the illustration could vary in size based on the content next to it. But that seems complex with low value.

> Task completed Icon

Oh yeah I never provided that sorry. I would reduce to 16dp. Thanks!

**Comment by AjeshRPai** [comment]
Hey @osullivanchris, Thanks for the review. I have done the following changes now. 

- [x] The heading text should be the same size as the 'Create your first/next post' heading text.
- [x] The progress bar would look better 1dp larger in height
- [x] 32dp padding between the left content area (heading, subheading, progress bar) and the illustration on the right
- [x] Task completed Icon - Reduced to 16 dp 

Regarding your other replies 
>54 width sounds right. Not sure I follow on the height. The original aspect ratio should be preserved so perhaps we should only define the width. For the vertical position, I was originally thinking centred to the content to the left. Maybe this is what you mean

Yup, I have maintained the same aspect ratio and the illustration is centred to the content on the left. 👍🏼

**Comment by AjeshRPai** [comment]
Hey @osullivanchris, In order to confirm 
❓ Is the design or specs of the card different in the home and menu tab?

**Comment by osullivanchris** [comment]
> ❓ Is the design or specs of the card different in the home and menu tab?

Sorry I missed that one! I think the screenshots of 'menu' are fine. I don't think we need to introduce two variations.

**Review by osullivanchris — CHANGES_REQUESTED** [review]
Thanks @AjeshRPai this is looking really great. All the changes look excellent.

One final detail I just want to get spot on. It looks like you took my idea about scaling the illustration. Did you make it totally variable to the height of the container or something? 

Here's what I'm seeing. Though technically I think what you're doing makes sense, when the string is shorter, its just making the card look a bit small and boring. The illustration is just going a bit too small I feel, and thus the height of the card as well. HEre's how it looks on my device for example
Screenshot_20220527-182310

I tried mocking up what I think looks right when there's a one line string or a two line string. There's two things I changed.

- On the left, the illustration is 40dp wide. On the right, its 56dp wide. I wonder if we could do the scaling like you have, but have a min or max value so that the card doesn't get too small or too large? Or maybe there's some other way do achieve this. 
- On the left (single line) the padding is increased to 24dp to balance the card. On the right its still 16dp. Another way to look at this would be to align the progress bar and subtitle to the bottom of the card vertically. But not sure if that is good practice.

Not sure how crazy these suggestions are. I do think this _looks right_, but I'm tweaking a lot just to make it look right in the case of one or two lines of text.
Screenshot 2022-05-27 at 18 35 32

**Review by osullivanchris — APPROVED** [review]
I'm gonna change my comment to approve. As I don't want you to be blocked merging this on my nitpick changes. You can get it into the build and then do my tweaks after or something if preferred.

**Comment by AjeshRPai** [comment]
@osullivanchris 

>On the left, the illustration is 40dp wide. On the right, its 56dp wide. I wonder if we could do the scaling like you have, but have a min or max value so that the card doesn't get too small or too large? Or maybe there's some other way do achieve this.

I think this can be done. Although I have some doubts regarding this implementation, I am going to take up this change in a separate PR and cc you in on it. 


@ashiagr As I have mentioned in the https://github.com/wordpress-mobile/WordPress-Android/pull/16629#issuecomment-1138262671, I have reverted the changes which were needed for the design review in the commit [285116d](https://github.com/wordpress-mobile/WordPress-Android/pull/16629/commits/285116de20a39d4932cae839c3bc6cd4b0753901). This PR is ready for another look. 👍🏼

**Review by ashiagr — APPROVED** [review]

**Comment by osullivanchris** [comment]
> I think this can be done. Although I have some doubts regarding this implementation, I am going to take up this change in a separate PR and cc you in on it.

Revisiting this with fresh eyes. There's two possibilities

1. We're gonna have lots of cards with illustrations. Maybe its useful to have some kinda dynamic scaling for images which work across different content. I do like what you did with the scaling (assuming this is what's happening) but to get it right I feel we need a min, max value. But having that dependent on the content length sounds pretty complex.
2. To make things much simpler I should just find a reasonable size that works pretty well no matter the content length to the left of it. While 40/56 either looked too large or too small depending on the content; I tried 48dp width and it looks reasonable in both cases. See img below. 

Should we just do the latter and have a set 48dp width?

Screenshot 2022-05-30 at 10 49 59

**Comment by AjeshRPai** [comment]
@osullivanchris 

>but to get it right, I feel we need a min, max value. But having that dependent on the content length sounds pretty complex

Yup, If there is a value in it. We can try this approach. It would be complex to vary the width according to the content length and keep a min/max value. I am afraid we might run into another scenario where it would not look correct.  

> I tried 48dp width and it looks reasonable in both cases. Should we do the latter and have a set 48dp width?

Yup, this looks good, to be honest. I say we will go with it 👍🏼 . Shall I go ahead and implement ?

**Comment by osullivanchris** [comment]
> Yup, this looks good, to be honest. I say we will go with it 👍🏼 . Shall I go ahead and implement ?

yeah lets do it! The value:effort of the other approach is questionable!

**Pull request merged**
